dScripts/AgSurvivalBuffStation.cpp

```diff
@@ -0,0 +1,19 @@
+#include "AgSurvivalBuffStation.h"
+#include "SkillComponent.h"
+#include "dLogger.h"
+
+void AgSurvivalBuffStation::OnRebuildComplete(Entity* self, Entity* target) {
+    auto skillComponent = self->GetComponent<SkillComponent>();
+
+    if (skillComponent == nullptr) return;
+
+    skillComponent->CalculateBehavior(201, 1784, self->GetObjectID());
+    
+    self->AddTimer("DestroyAfter10Seconds", 10.0f);
```

Comment:
You can also use a callback timer here

Reply:
(Non blocking)

Reply:
Changed this out for a callback timer.  I dont understand _how_ exactly they work yet but I followed what a callback timer looks like for other scripts and that seemed to work.